usaspending_api/awards/management/commands/loadcontracts.py

```diff
@@ -26,25 +27,227 @@ def handle(self, *args, **options):
+            "subcontracting_plan": lambda row: row['subcontractplan'].split(' ')[0].split(':')[0],
+            "type_set_aside": lambda row: row['typeofsetaside'].split(' ')[0].split(':')[0],
+            "walsh_healey_act": lambda row: row['walshhealyact'].split(' ')[0].split(':')[0],
+            "rec_flag": lambda row:  self.parse_first_character(row['rec_flag'].split(' ')[0].split(':')[0]),
+            "type_of_idc": lambda row: self.parse_first_character(row['typeofidc']),
+            "a76_fair_act_action": lambda row: self.parse_first_character(row['a76action']),
+            "clinger_cohen_act_planning": lambda row: self.parse_first_character(row['clingercohenact']),
+            "cost_accounting_standards": lambda row: self.parse_first_character(row['costaccountingstandardsclause']),
+            "fed_biz_opps": lambda row: self.parse_first_character(row['fedbizopps']),
+            "foreign_funding": lambda row: self.parse_first_character(row['fundedbyforeignentity']),
+            "major_program": lambda row: self.parse_first_character(row['majorprogramcode']),
+            "program_acronym": lambda row: self.parse_first_character(row['programacronym']),
+            "referenced_idv_modification_number": lambda row: self.parse_first_character(row['idvmodificationnumber']),
+            "transaction_number": lambda row: self.parse_first_character(row['transactionnumber']),
+            "solicitation_identifier": lambda row: self.parse_first_character(row['solicitationid']),
             "submission": subattr
         }
 
         loader = ThreadedDataLoader(Procurement, field_map=field_map, value_map=value_map)
         loader.load_from_file(options['file'][0])
 
+    def evaluate_contract_award_type(self, row):
+        first_element = row['contractactiontype'].split(' ')[0].split(':')[0]
+        if len(first_element) == 1:
+            return first_element
+        else:
+            cat = row['contractactiontype'].lower()
+            # Not using DAIMS enumeration . . .
+            if 'bpa' in cat:
+                return 'A'
+            if 'purchase' in cat:
+                return 'B'
+            elif 'delivery' in cat:
+                return 'C'
+            elif 'definitive' in cat:
+                return 'D'
+            else:
+                return None
+
+    def get_agency(self, row):
+        agency = Agency.objects.filter(subtier_code=self.get_agency_code(row['maj_agency_cat'])).first()
+        if not agency:
+            print("Missing agency: " + row['maj_agency_cat'])
+        return agency
+
+    def create_or_get_recipient(self, row):
+        # First, create the locations
+        location_dict = {}
+        pop_dict = {}
+
+        loc_country_code = row["vendorcountrycode"].split(":")[0]
+
+        loc_country_code = RefCountryCode.objects.filter(Q(**{"country_code": loc_country_code}) | Q(**{"country_name__iexact": loc_country_code})).first()
+        if not loc_country_code:
```

Comment:
Yeah, for clarification of anyone coming behind to read:
Sometimes "vendorcountrycode" isn't a country code at all, sometimes it is a name. Like "SOUTH KOREA" instead of "KOR". However, our reference table stores countries in patterns like "KOREA, SOUTH" or "BAHAMAS, THE". Thus, I chain together filters on each word to be able to locate the country code by name if there is no direct match to name or country code.